guacamole-common-js/src/main/webapp/modules/Keyboard.js

```diff
@@ -1381,24 +1380,37 @@ Guacamole.Keyboard = function Keyboard(element) {
             if (!markEvent(e)) return;
 
             // Type all content written
-            if (e.data && !e.isComposing) {
-                element.removeEventListener("compositionend", handleComposition, false);
+            if (e.data && !e.isComposing)
                 guac_keyboard.type(e.data);
-            }
+
+        };
+
+        /**
+         * Handles the given "compositionstart" event, automatically removing
+         * the "input" event handler, as "input" events should only be handled
+         * IFF composition events are not provided by the browser.
```

Comment:
IFF = ?

Maybe I'm just getting old and don't understand all the new-fangled language ;-).

Reply:
Oh, If and Only If. Though, saying "only If and Only If" is awfully redundant... let me fix that.

Reply:
I suspected it might be that, but not terribly familiar with it...